resharper/resharper-unity/src/Rider/RiderPackageUpdateAvailabilityChecker.cs

```diff
@@ -0,0 +1,127 @@
+using JetBrains.Rider.Model.Notifications;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Rider
+{
+    [SolutionComponent]
+    public class RiderPackageUpdateAvailabilityChecker
+    {
+        private readonly ILogger myLogger;
+        private readonly NotificationsModel myNotifications;
+        private readonly ISolution mySolution;
+        private readonly IShellLocks myShellLocks;
+        private readonly PackageManager myPackageManager;
+        private readonly UnityVersion myUnityVersion;
+        private readonly ISettingsStore mySettingsStore;
+        private readonly JetHashSet<VirtualFileSystemPath> myNotificationShown;
+        private readonly IContextBoundSettingsStoreLive myBoundSettingsStore;
+        private string packageId = "com.unity.ide.rider";
+        private string myHyperlinkId = "NeverShowRiderPackageUpdateAvailable";
+
+        public RiderPackageUpdateAvailabilityChecker(
+            Lifetime lifetime,
+            ILogger logger,
+            NotificationsModel notifications,
+            ISolution solution,
+            IShellLocks shellLocks,
+            PackageManager packageManager,
+            UnitySolutionTracker unitySolutionTracker,
+            UnityVersion unityVersion,
+            IApplicationWideContextBoundSettingStore applicationWideContextBoundSettingStore,
+            ISettingsStore settingsStore
+            )
+        {
+            myLogger = logger;
+            myNotifications = notifications;
+            mySolution = solution;
+            myShellLocks = shellLocks;
+            myPackageManager = packageManager;
+            myUnityVersion = unityVersion;
+            mySettingsStore = settingsStore;
+            myNotificationShown = new JetHashSet<VirtualFileSystemPath>();
+            myBoundSettingsStore = applicationWideContextBoundSettingStore.BoundSettingsStore;
+            unitySolutionTracker.IsUnityProjectFolder.WhenTrue(lifetime, lt =>
+            {
+                ShowNotificationIfNeeded(lt);
+                BindToInstallationSettingChange(lt);
+            });
+        }
+        
+        private void BindToInstallationSettingChange(Lifetime lifetime)
+        {
+            var entry = myBoundSettingsStore.Schema.GetScalarEntry((UnitySettings s) => s.AllowRiderUpdateNotifications);
+            myBoundSettingsStore.GetValueProperty<bool>(lifetime, entry, null).Change.Advise_NoAcknowledgement(lifetime, args =>
+            {
+                if (!args.GetNewOrNull()) return;
+                ShowNotificationIfNeeded(lifetime);
+            });
+        }
+
+        private void ShowNotificationIfNeeded(Lifetime lifetime)
+        {
+            if (!myBoundSettingsStore.GetValue((UnitySettings s) => s.AllowRiderUpdateNotifications))
+                return;
+            
+            myPackageManager.IsInitialUpdateFinished.WhenTrue(lifetime, lt =>
+            {
+                var version = myUnityVersion.ActualVersionForSolution.Value;
```

Comment:
Might be worth combining `ActualVersionForSolution` with `IsInitialUpdateFinished`. When both are true/valid, then run the check